app/helpers/solidus_sale_prices/soft_delete.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal :true
+
+module SolidusSalePrices
+  module SoftDelete
+    def self.delete_method
```

Comment:
I'm not sure we need this. What if we just always use Discard methods? If I recall correctly, in Solidus 2.10 we had support for both Discard and Paranoia just for backward compatibility, so using discard methods would work exactly the same without emitting deprecation warnings. Did you already try?

Reply:
In Solidus 2.10, Spree::SoftDeletable has not been added yet. This prevents the use of discard methods which is the reason for the conditional methods and I thought it might be cleaner to but some of that logic in a helper

Reply:
Even without `Spree::SoftDeletable`, discard is still present. I don't see any need to use paranoia.

Reply:
I will explore this further, I was under the impression that the functionality of discard was tied to Spree::SoftDeletable

Reply:
Just started playing with it this morning and found that the specs do fail with just the discard methods implemented. It works and is arguably more concise than the previous implementation if I just force the SoftDeleteable functionality in the price model:
```
    if ::Spree.solidus_gem_version >= Gem::Version.new('2.11')
      include Spree::SoftDeletable
    else
      require 'discard'
      acts_as_paranoid
      include Discard::Model
      self.discard_column = :deleted_at
    end
```
Further, I could throw that into the helper module and place a note to remove when 2.10 is no longer supported as well as create an issue in this repository.

Additionally, without paranoia, the following fails and requires further investigation:
```
  1) Spree::Variant with a valid sale destroys all sale prices when it is destroyed
     Failure/Error:
       expect { variant.discard }
         .to change { Spree::SalePrice.all.size }
         .from(3).to(0)
     
       expected `Spree::SalePrice.all.size` to have changed from 3 to 0, but did not change
     # ./spec/models/variant_spec.rb:144:in `block (3 levels) in <top (required)>'

  2) Spree::Price#destroy when there are sale prices associated to the price destroys all sale prices when it is destroyed
     Failure/Error:
       expect { price.discard }
         .to change { Spree::SalePrice.all.size }
         .from(1).to(0)
     
       expected `Spree::SalePrice.all.size` to have changed from 1 to 0, but did not change
     # ./spec/models/price_spec.rb:92:in `block (4 levels) in <top (required)>'
```
Without Discard::Model, none of the discard methods work.

I will continue to look for other solutions and please let me know if I am steering off in the wrong direction.
Thanks!